homeassistant/components/device_tracker/setup.py

```diff
@@ -176,13 +178,16 @@ async def async_device_tracker_scan(now: dt_util.dt.datetime):
                 "mac": mac,
                 "host_name": host_name,
                 "source_type": SOURCE_TYPE_ROUTER,
+                "location_name": location_name,
                 "attributes": {
                     "scanner": scanner.__class__.__name__,
                     **extra_attributes,
                 },
             }
 
-            zone_home = hass.states.get(hass.components.zone.ENTITY_ID_HOME)
+            zone_home = hass.states.get(
+                hass.components.zone.ENTITY_ID_FORMAT.format(location_name)
+            ) or hass.states.get(hass.components.zone.ENTITY_ID_HOME)
```

Comment:
Is it worth renaming this to something like current_zone to reflect that we've seen it anywhere rather than in zone.home?